openlibrary/plugins/upstream/covers.py

```diff
@@ -73,14 +74,16 @@ def upload(self, key, i):
         try:
             if six.PY3:
                 files = {'data': BytesIO(data)}
-                resp = requests.post(upload_url, data=params, files=files)
+                response = requests.post(upload_url, data=params, files=files)
             else:
                 params['data'] = data
                 payload = requests.compat.urlencode(params).encode('utf-8')
-                resp = requests.post(upload_url, data=payload)
-            return web.storage(resp.json())
+                response = requests.post(upload_url, data=payload)
+            response.raise_for_status()
```

Comment:
Might want to drop this line in case /upload2 returns a useful response with a not ok status code.